File: Arrays/Python/Valid_Pair_Sum.py
class Solution:
    def ValidPair(self, a, n): 
    	# Your code goes here
    	a.sort(reverse = True) # sort array
        low, high = 0, n-1
        count = 0
        while low <= high:
            msum = a[low] + a[high]
            if msum <= 0:
                high -= 1
            else:
                count += high -low
                low += 1
        return count

File: Arrays/Java/Valid_Pair_Sum.java
class Solution 
{ 
    static long ValidPair(int a[], int n) 
	{ 
	    // Your code goes here
	    Arrays.sort(a);
        long count=0;
        int i=0;
        int j=n-1;
        while(i<j) {
            if(a[i]+a[j]<=0) {
                i++;
            }
            else { 
                count+=j-i;
                j--;
            }
        }
        return count;
	}
} 
